sdk/python/tests/compiler/compiler_tests.py

```diff
@@ -99,6 +114,9 @@ def _test_pipeline_workflow(self, pipeline_function, pipeline_yaml, generate_pip
     try:
       compiler.TektonCompiler().compile(pipeline_function, compiled_yaml_file, generate_pipelinerun=generate_pipelinerun)
       with open(compiled_yaml_file, 'r') as f:
+        if 'loop' in pipeline_yaml:
```

Comment:
would be great if we could wrap this transformation into a `lambda` function and pass a `normalize_compiler_output` function as a parameter to `_test_pipeline_workflow`

That way the `_test_pipeline_workflow` method could remain general and other tests which need to modify the generated "YAML" in order to compare with "golden" output could also pass transformations.